File: src/automatica.core/Automatica.Core.Model/Models/User/Role.cs
using System;
using System.Collections.Generic;

namespace Automatica.Core.Model.Models.User
{
    public class Role : TypedObject
    {
        public const string AdminRole = "administrator";
        public const string ViewerRole = "viewer";
        public const string VisuRole = "visu";


        public Role()
        {
            InverseThis2Priviledges = new List<Priviledge2Role>();
        }

        public Guid ObjId { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }

        public string Key { get; set; }

        public bool IsDeleteable { get; set; }

        public List<Priviledge2Role> InverseThis2Priviledges { get; set; }

    }
}

File: src/automatica.core/Automatica.WebNew/src/app/base/model/user/role.ts
import { BaseModel, Model, JsonFieldInfo, JsonProperty } from "../base-model";
import { INameModel } from "../INameModel";
import { IDescriptionModel } from "../IDescriptionModel";

@Model()
export class Role extends BaseModel implements INameModel, IDescriptionModel {
    public static ADMIN_ROLE = "administrator";
    public static VISU_ROLE = "visu";

    @JsonProperty()
    Key: string;

    @JsonProperty()
    Name: string;

    @JsonProperty()
    Description: string;

    @JsonProperty()
    ObjId: string;

    @JsonProperty()
    InverseThis2Roles: any[];

    public get DisplayName() {
        return this.translationService.translate(this.Name);
    }

    public get DisplayDescription() {
        return this.translationService.translate(this.Description);
    }

    public typeInfo(): string {
        return "Role";
    }

    protected getJsonProperty(): Map<string, JsonFieldInfo> {
        return new Map<string, JsonFieldInfo>();
    }
}
